tests/Middleware/ActionCallerTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Middleware;
+
+use Nyholm\Psr7\Response;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Yii\Web\Middleware\ActionCaller;
+use PHPUnit\Framework\TestCase;
+
+class ActionCallerTest extends TestCase
+{
+    public function testProcess()
+    {
+        $container = $this->createMock(ContainerInterface::class);
+        $container->method('get')->willReturn($this);
+
+        $request = $this->createMock(ServerRequestInterface::class);
+        $handler = $this->createMock(RequestHandlerInterface::class);
+
+        $caller = new ActionCaller('this', 'exampleMethod', $container);
+
+        $response = $caller->process($request, $handler);
+        self::assertEquals(204, $response->getStatusCode());
+    }
+
+    public function exampleMethod(): ResponseInterface
```

Comment:
Without passing to exampleMethod `assertEquals(204, $response->getStatusCode());` fails, because not default 204 response code sets in exampleMethod(). DI returns ActionCallerTest itself and ActionCaller calls 'exampleMethod' in ActionCallerTest instance.

Should I rewrite all on mocks without using "helper" methods of test classes?

Reply:
I've meant checking that `$request` object is the same instance as was mocked and that `$test` is an instance of `CustomDependency` obtained from container:

```php
public function exampleMethod(ServerRequestInterface $request, CustomDependency $test): ResponseInterface
{
   // ...
}
```

Reply:
> Should I rewrite all on mocks without using "helper" methods of test classes?

The less mocks the better.